Assets/Mirror/Migration/Editor/Scripts.cs

```diff
@@ -132,12 +131,16 @@ public class Scripts : MonoBehaviour {
             StreamWriter sw;
 
             foreach (string file in filesToProcess) {
-                try {
+                try
+                {
+                    Encoding encoding = Utils.GetEncoding(file);
```

Comment:
Detect the encoding from BOM if file has it, then we can save with the same encoding